ext/score.c

```diff
@@ -0,0 +1,45 @@
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <openssl/sha.h>
+
+// This function converts the input data into SHA-256 hash
+static unsigned char* sha256(unsigned char* hash, char* data) {
+  size_t length = strlen((char*)data);
+  SHA256((const unsigned char*)data, length, hash);
+  return hash;
+}
+
+// Calculate nonce for strength x with a given data, offset and limit
+uint64_t calculate_nonce_extended(uint64_t offset, uint64_t limit, char* data, uint8_t strength) {
+  uint64_t nonce = offset;
+  size_t data_length = strlen(data);
+  char* current_data = malloc(data_length + 20);
+  unsigned char* hash = malloc(SHA256_DIGEST_LENGTH);
+  char* current_data_nonce = current_data + data_length;
+  memcpy(current_data, data, data_length);
+  while(1) {
+    sprintf(current_data_nonce, "%lx", nonce);
+    hash = sha256(hash, current_data);
+    for(uint8_t i = 1; i <= strength / 2; i++) {
+      if(hash[SHA256_DIGEST_LENGTH - i] != 0) {
+        goto next_nonce_round;
+      }
+    }
+    if(strength & 1
+       && (hash[SHA256_DIGEST_LENGTH - (strength / 2) - 1] & 0x0f) != 0) {
+      goto next_nonce_round;
+    }
+    break;
+next_nonce_round:
```

Comment:
Sincerely, I don't like very much labels and goto statement as they make difficult to trace the flow of the method

Reply:
Again logoc over style. This code is for optimization not beauty.

Reply:
Code readbility is not beauty but it has to do with readibility and maintanability too. It is important that developers provide with soultions that are easy to read and mantain for others developers. Just google "goto statement" and you will see tons of articles discouraging the use of goto and labels, for these same reasons.

Also, removing the label would be very easy, just matter of using a flag variable to exit the loop.

Reply:
@oltdaniel I see you haven't removed the goto statement, in my opinion that should be done for the sake of maintainability but maybe we should leave the final decision about it to @yegor256

Reply:
@yegor256 ping